torch/nn/parallel/distributed.py

```diff
@@ -100,9 +100,20 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
         self.device_ids = device_ids
         self.output_device = output_device
 
+        MB = 1024 * 1024
+        # used for intra-node param sync and inter-node sync as well
+        self.broadcast_bucket_size = 10 * MB
+
         # Sync params and buffers
-        for p in self.module.state_dict().values():
-            dist.broadcast(p, 0)
+        # Here we will coalesce all the parameters and buffers in several big
+        # flat tensors and broadcast them out to reduce the number of broadcasts
+        # as well as improve performance, even though this function is
+        # only called one time.
```

Comment:
I don't think this comment is necessary. The important part is that we have to sync params. How we do it is clear from the code (these are just 2 lines).